### PR TITLE
S4 add review

### PR DESCRIPTION
**Pull request by PapaPwn**
Everything seems to be working, except state is stale when asking repeated questions, such that it contains previous question's answers. Not sure how to fix...

### CODE REVIEW

**Comment by NunoDasNeves** [comment]
I think this is all g

File: frontend/src/components/questions-answers/QuestionCard.vue [hunk 1]
```diff
@@ -7,7 +7,7 @@
                 <p class="vote">&minus;</p>
             </div>
             <div class="content">
-                <router-link tag="h2" :to="{ name: 'question', param: { id: String(id) }}">
+                <router-link tag="h2" :to="{ name: 'question', params: { id: String(id) }}">
```
Comment (DarkPurple141) [inline frontend/src/components/questions-answers/QuestionCard.vue]: it is params right?

Comment (DarkPurple141) [reply frontend/src/components/questions-answers/QuestionCard.vue]: Yes it is.

File: frontend/src/router.js [hunk 2]
```diff
@@ -38,11 +38,19 @@ export default new Router({
...
+        courseID: String(cid)
+      }),
+      component: () => import('./views/Review')
+    },
     {
       path: '/review/:id',
       name: 'review',
       props: ({params: { id }}) => ({
-        id
+        reviewID: String(id)
```
Comment (DarkPurple141) [inline frontend/src/router.js]: Feel like if we're doing the cast here then it should be impossible for it ever to not be cast?

File: frontend/src/views/Review.vue [hunk 3]
```diff
@@ -11,24 +34,67 @@
...
+    ReplyForm
   },
   props: {
-    id: String
+    courseID: String, // This is a query
+    reviewID: String // This is a param
+  },
+  data() {
+    return {
+      commentType: 'Reply'
```
Comment (DarkPurple141) [inline frontend/src/views/Review.vue]: ?

File: frontend/src/views/Review.vue [hunk 4]
```diff
@@ -11,24 +34,67 @@
...
+        // this.answerFormResponse.text = "Please type an answer!"
+        // this.answerFormResponse.style = {'form-success': false, 'form-failure': true}
+        return
+      }
+      this.$store.dispatch('reviews/postReply', {form: replyForm, id: this.review.id})
+    }
+  },
   created () {
-    this.$store.dispatch('reviews/getReview', this.id)
+    if (this.reviewID) {
```
Comment (DarkPurple141) [inline frontend/src/views/Review.vue]: This should be done in beforeRouteUpdate.

File: frontend/src/App.vue [hunk 5]
```diff
@@ -53,6 +53,7 @@ html, body {
   /* arbitrary */
   min-height: calc(100vh - 150px);
   max-width: 768px;
+  min-height: 100vh;
```
[the remark this answers is not in the document]
Comment (NunoDasNeves) [reply frontend/src/App.vue]: this looks like a bad merge...

File: frontend/src/components/comments/CommentForm.vue [hunk 6]
```diff
@@ -1,11 +1,11 @@
 <template>
     <Card>
       <div class="content">
-          <h3>Submit an Answer</h3>
+          <h3>Submit {{ type }}</h3>
           <p>
-            <textarea placeholder="Your answer here.." v-model="body"></textarea><br>
+            <textarea placeholder="Your input here.." v-model="body"></textarea><br>
             <!-- should probs be a separate component -->
-            <AppButton @click.native="$emit('submitAnswerForm', {body})">Answer</AppButton>
```
[the remark this answers is not in the document]
Comment (NunoDasNeves) [reply frontend/src/components/comments/CommentForm.vue]: oh yeah that's better

File: frontend/src/views/Review.vue (hunk 3 above)
Comment (PapaPwn) [reply frontend/src/views/Review.vue]: This is used as the text for the button and comment prompt, i.e. it is currently either set to "Reply" or "Answer"

File: frontend/src/components/course/Feed.vue [hunk 7]
```diff
@@ -23,6 +23,9 @@ export default {
   props: {
     feedType: String,
     items: Array
+  },
+  mounted() {
+    console.log('mounting feed')
```
[the remark this answers is not in the document]
Comment (DarkPurple141) [reply frontend/src/components/course/Feed.vue]: remove.

[the remark this answers is not in the document]
Comment (DarkPurple141) [reply frontend/src/components/course/Feed.vue]: Also feel free to add same logic in questions for transitions to reviews.

**Pull request merged**